### PR TITLE
Onfleet errors

### PR DESCRIPTION
**Pull request by erickreutz**
Adds error classes to handle Onfleet's error responses

### CODE REVIEW

**Comment by erickreutz** [comment]
@pedro what do you think about this?

File: lib/onfleet/resource.rb [hunk 1]
```diff
@@ -96,13 +99,85 @@ def self.delete(action = "", values = {})
...
-      response = Nestful::Request.new(
-        uri, options
-      ).execute
+      begin
+        response = Nestful::Request.new(
+          uri, options
+        ).execute
+      rescue JSON::ParserError
+        raise Errors::Error
+      rescue Nestful::ResponseError => e
```
Comment (pedro) [inline lib/onfleet/resource.rb:108]: I'd rescue all `Nestful::Error` instances here (or in a diff rescue clause), as it includes some [other weird connection issues](https://github.com/maccman/nestful/blob/master/lib/nestful/exceptions.rb#L11) that might show up.

File: lib/onfleet/resource.rb [hunk 2]
```diff
@@ -96,13 +99,85 @@ def self.delete(action = "", values = {})
...
+
+      case response.code
+      when "400"
+        raise create_error Errors::InvalidContent, *err_args
+      when "401"
+        raise create_error Errors::InvalidCredentials, *err_args
+      when "404"
+        raise create_error Errors::ResourceNotFound, *err_args
+      when "409"
+        raise create_error Errors::PreconditionFailed, *err_args
```
Comment (pedro) [inline lib/onfleet/resource.rb]: I think there's a typo here, should it be a `InvalidArgumentError`?

File: lib/onfleet/resource.rb [hunk 3]
```diff
@@ -96,13 +99,85 @@ def self.delete(action = "", values = {})
...
+      when "500"
+        raise create_error Errors::InternalError, *err_args
+      else
+        raise create_error Errors::OnfleetError, *err_args
+      end
+    end
+
+    def self.general_error(response_code, response_body)
+      Errors::Error.new(
+        message: "Invalid response object from API: #{response_code.inspect} " + "(HTTP response code was #{response_code})",
```
Comment (pedro) [inline lib/onfleet/resource.rb]: Response code shows up twice, maybe shorten to just "Onfleet API failed with #{response_code}"?

**Comment by pedro** [comment]
@erickreutz nitpicked some stuff above but this looks awesome :+1::+1::+1:

File: lib/onfleet/resource.rb (hunk 3 above)
Comment (erickreutz) [inline lib/onfleet/resource.rb]: It should actually be response_body.inspect my bad.

**Comment by erickreutz** [comment]
Thanks man! I made the changes. Anything else you think before we merge this in?

**Comment by pedro** [comment]
Nothing, good stuff!

**Pull request merged**